tools/congestion-model/src/model/mod.rs

```diff
@@ -35,26 +38,44 @@ pub struct Model {
     // Workload state
     pub(crate) transactions: TransactionRegistry,
     pub(crate) producer: Box<dyn Producer>,
+
+    stats_writer: StatsWriter,
+    start_time: DateTime<Utc>,
 }
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct ShardId(usize);
+pub struct ShardId(pub usize);
 
 impl Model {
     pub fn new(
         mut shards: Vec<Box<dyn CongestionStrategy>>,
         mut producer: Box<dyn Producer>,
+        mut stats_writer: StatsWriter,
     ) -> Self {
+        // Set the start time to an hour ago to make it visible by default in
+        // grafana. Each round is 1 virtual second so 1 hour is good for looking
+        // at a maximum of 3600 rounds, beyond that you'll need to customize the
+        // grafana time range.
+        let start_time = Utc::now() - Duration::from_secs(1 * 60 * 60);
         let num_shards = shards.len();
         let shard_ids: Vec<_> = (0..num_shards).map(ShardId).collect();
         let mut queues = QueueBundle::new(&shard_ids);
 
+        if let Some(stats_writer) = &mut stats_writer {
+            stats_writer.write_field("time").unwrap();
+            stats_writer.write_field("round").unwrap();
+        }
+
```

Comment:
this code should be somewhere in `.src/evaluation` and be called by the model runner in main.rs